crates/uv-resolver/src/prerelease_mode.rs

```diff
@@ -95,4 +95,15 @@ impl PreReleaseStrategy {
             ),
         }
     }
+
+    /// Returns `true` if a [`PackageName`] is allowed to have pre-release versions.
+    pub(crate) fn allows(&self, package: &PackageName) -> bool {
+        match self {
+            Self::Disallow => false,
+            Self::Allow => true,
+            Self::IfNecessary => false,
```

Comment:
There's a potential bug here. If a package has _only_ pre-releases, we _will_ allow it in the candidate selector (e.g., how `black` behaved long ago). However... we won't know about it here (since we don't know this information in advance). I wonder if we should drop support for this.

Reply:
FYI, if you need a real world use case, `opentelemetry-exporter-prometheus` only has pre-releases, and `apache-airflow[otel]` depends on this (and hence also `apache-airflow[all]`).

pip very much does have this behavior of including packages that only have prereleases into a regular install.